app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -262,7 +274,8 @@ function FormSteps({
   }, [stepErrors]);
 
   useStepTitle(step, titleFormat);
-  useDidUpdateEffect(onStepChange, [step]);
+  trackStepEvents(step);
+  useDidUpdateEffect(onStepChange.bind(this, step), [step]);
```

Comment:
I think in most cases where there's a `#bind`, we can probably just use an arrow function instead. Similarly compact, but a bit more clear:

```suggestion
  useDidUpdateEffect(() => onStepChange(step), [step]);
```